modules/nf-core/fqtk/main.nf

```diff
@@ -0,0 +1,47 @@
+process FQTK {
+    tag "$meta.id"
+    label 'process_high'
+
+    conda "bioconda::fqtk=0.2.0"
+    container "${ workflow.containerEngine == 'singularity' && !task.ext.singularity_pull_docker_container ?
+        'https://depot.galaxyproject.org/singularity/fqtk:0.2.0--h9f5acd7_0' :
+        'quay.io/biocontainers/fqtk:0.2.0--h9f5acd7_0' }"
+
+    input:
+    tuple val(meta), path(sample_sheet), path(fastq_dir)
+    val(fastq_readstructure_pairs)
```

Comment:
add doc for what the expected structure is?

```
// List[(?, ?, ?)]
```

Reply:
I think there needs to be some thought about what the inputs to this module is, and how it is best provided by the nf-core/demultiplex pipeline.  Can you describe your plan for how this is called in the latter?

Reply:
I'll put my hat in the ring.  I think it should take in

```nextflow
input:
tuple val(meta), path(sample_sheet), val (read_structures_and_fastqs)
```

where `read_structure_and_fastqs` is a list of tuples, with each tuple being a read structure and the path to the associated FASTQ

I prefer this over providing a `fastq_dir`, since `fqtk` does not take a fastq directory as input, but instead the read FASTQ files explicitly.  And since the read structures and FASTQs have to be provided in the same order, it makes sense to keep them paired as tuples. 

A tempting alternative is:

```nextflow
input:
tuple val(meta), path(sample_sheet)
val (read_structures_and_fastqs)
```

but that makes less sense to me, since that indicates you could fix one input and vary the other (e.g. use the same meta/sample sheet but have different list of read-structure/fastq tuples).  I think that doesn't make a lot of sense.

Reply:
@nh13 At first glance I like this suggestion, I am trying to implement this now.

Reply:
Here is my fist shot at your suggestion 6815a006e92d04894fcba02fa91355ff550872c2

Reply:
LGTM

Reply:
@nh13 Still working on re-working to be from params not config csv